src/test/java/uk/gov/ons/ctp/integration/rhsvc/endpoint/CaseEndpointUnitTest.java

```diff
@@ -65,46 +61,41 @@ public void setUp() {
   public void getCaseByUPRNFound() throws Exception {
     List<CaseDTO> caseDTO = FixtureHelper.loadClassFixtures(CaseDTO[].class);
     CaseDTO rmCase0 = caseDTO.get(0);
+    CaseDTO rmCase1 = caseDTO.get(1);
 
-    when(caseService.getLatestValidCaseByUPRN(new UniquePropertyReferenceNumber(UPRN)))
-        .thenReturn(caseDTO.get(0));
+    when(caseService.readCaseUpdateBySampleAttribute("uprn", UPRN)).thenReturn(caseDTO);
 
     mockMvc
-        .perform(get("/cases/uprn/{uprn}", UPRN))
+        .perform(get("/cases/attribute/uprn/{uprn}", UPRN))
         .andExpect(status().isOk())
         .andExpect(content().contentType(EXPECTED_JSON_CONTENT_TYPE))
-        // .andExpect(jsonPath("$", hasSize(1)))
-        .andExpect(jsonPath("$.caseId", is(rmCase0.getCaseId().toString())))
-        .andExpect(jsonPath("$.address.addressLine1", is(rmCase0.getAddress().getAddressLine1())))
-        .andExpect(jsonPath("$.address.townName", is(rmCase0.getAddress().getTownName())))
-        .andExpect(jsonPath("$.address.postcode", is(rmCase0.getAddress().getPostcode())));
+        .andExpect(jsonPath("$.length()", is(2)))
+        .andExpect(jsonPath("$[0].caseId", is(rmCase0.getCaseId().toString())))
+        .andExpect(
+            jsonPath("$[0].sample.addressLine1", is(rmCase0.getSample().get("addressLine1"))))
+        .andExpect(jsonPath("$[0].sample.townName", is(rmCase0.getSample().get("townName"))))
+        .andExpect(jsonPath("$[0].sample.postcode", is(rmCase0.getSample().get("postcode"))))
+        .andExpect(jsonPath("$[1].caseId", is(rmCase1.getCaseId().toString())))
+        .andExpect(
+            jsonPath("$[1].sample.addressLine1", is(rmCase1.getSample().get("addressLine1"))))
+        .andExpect(jsonPath("$[1].sample.townName", is(rmCase1.getSample().get("townName"))))
+        .andExpect(jsonPath("$[1].sample.postcode", is(rmCase1.getSample().get("postcode"))));
   }
 
   /** Test returns resource not found for non-existent UPRN */
   @Test
   public void getCaseByUPRNNotFound() throws Exception {
 
-    when(caseService.getLatestValidCaseByUPRN(new UniquePropertyReferenceNumber(UPRN)))
+    when(caseService.readCaseUpdateBySampleAttribute("uprn", UPRN))
         .thenThrow(new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND, ERROR_MESSAGE));
 
     mockMvc
-        .perform(get("/cases/uprn/{uprn}", UPRN))
+        .perform(get("/cases/attribute/uprn/{uprn}", UPRN))
         .andExpect(status().isNotFound())
         .andExpect(jsonPath("$.error.code", is(CTPException.Fault.RESOURCE_NOT_FOUND.toString())))
         .andExpect(jsonPath("$.error.message", is(ERROR_MESSAGE)));
   }
 
```

Comment:
consider adding another test where a different attribute other than "uprn" is used.

Reply:
Have modified one of the tests to use a different attribute.